homeassistant/components/coolmaster_serial/climate.py

```diff
@@ -0,0 +1,202 @@
+"""coolmaster_serial platform to control of CoolMaste Climate Devices."""
+
+import logging
+
+from pycoolmaster import CoolMaster
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    HVAC_MODE_COOL,
+    HVAC_MODE_DRY,
+    HVAC_MODE_FAN_ONLY,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_HEAT_COOL,
+    HVAC_MODE_OFF,
+    SUPPORT_FAN_MODE,
+    SUPPORT_SWING_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    PRECISION_WHOLE,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+
+from .const import AVAILABLE_MODES, DOMAIN, CONF_SERIAL_PORT, CONF_BAUDRATE
+
+SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE  # | SUPPORT_SWING_MODE
```

Comment:
I'm seeking guidance on this...  Not all systems support reading the swing mode.  In the communications library, I set swing mode to None if it can't be read (https://github.com/issacg/pycoolmaster/blob/master/pycoolmaster/__init__.py#L148), but I started getting a whole slew of exceptions from HomeAssistant when I left this on.

In order to minimize friction in getting the initial integration in, I'd be more than happy to drop the SUPPORT_SWING_MODE for now, and try to deal with it more elegantly later, but I'm happy to get any feedback from reviewers.

Thanks!